yamleditor-plugin/src/main/java-eclipse/de/jcup/yamleditor/ReplaceTabBySpacesVerifyKeyListener.java

```diff
@@ -59,33 +62,125 @@ public void run() {
 					if (offset == -1) {
 						offset = yamlEditor.lastCaretPosition;
 					}
+
+					boolean isMultiline = ts.getStartLine() != -1 && ts.getEndLine() > ts.getStartLine();
+					boolean doIndent = event.stateMask == 0;
+					boolean doOutdent = (event.stateMask & SWT.SHIFT) == SWT.SHIFT;
+					if (!doIndent && !doOutdent)
+						return;
+
 					try {
-						int spaces = YamlEditorPreferences.getInstance().getAmountOfSpacesToReplaceTab();
-						String toInsert = createTabReplacement(spaces);
-						int toInsertLength = toInsert.length();
-						doc.replace(offset, ts.getLength(), toInsert);
-						Control control = yamlEditor.getAdapter(Control.class);
-						if (control instanceof StyledText) {
-							StyledText t = (StyledText) control;
-							t.setCaretOffset(offset + toInsertLength);
+						int numSpaces = YamlEditorPreferences.getInstance().getAmountOfSpacesToReplaceTab();
+						if (numSpaces < 1)
+							return;
+
+						String tabReplacement = createTabReplacement(numSpaces);
+
+						if (isMultiline) {
+							// in case there is a multiline selection, we mimic the Eclipse behavior
+							// (which is quite standard across editors) and thus we:
```

Comment:
I will move such long code into dedicated methods, so its easier to understand whats going on.